src/librustc_lexer/src/lib.rs

```diff
@@ -249,7 +274,8 @@ pub fn first_token(input: &str) -> Token {
 }
 
 /// Creates an iterator that produces tokens from the input string.
-pub fn tokenize(mut input: &str) -> impl Iterator<Item = Token> + '_ {
+pub fn tokenize(input: &str) -> impl Iterator<Item = Token> + '_ {
+    let mut input = input;
```

Comment:
Nit: unnecessary change?

Reply:
without this change, I couldn't call it with a str that isn't mut, but maybe I messed something up